### PR TITLE
Introduce the "Relayable" entity

### PR DESCRIPTION
**Pull request by cmrd-senya**
Introduce the "Relayable" entity, which encapsulates common properties of entities that are inherited from Relayable class in the Diaspora source code - "parent_guid", "parent_author_signature", "author_signature".

### CODE REVIEW

File: spec/lib/diaspora_federation/validators/comment_validator_spec.rb [hunk 1]
```diff
@@ -15,14 +15,6 @@ module DiasporaFederation
...
     end
 
-    %i(author_signature parent_author_signature).each do |prop|
-      describe "##{prop}" do
-        it_behaves_like "a property that mustn't be empty" do
-          let(:property) { prop }
-        end
-      end
-    end
-
```
Comment (SuperTux88) [inline spec/lib/diaspora_federation/validators/comment_validator_spec.rb]: Please add a `it_behaves_like "a relayable validator"` where you still test this.

File: lib/diaspora_federation/validators/relayable_validator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module DiasporaFederation
+  module Validators
+    module RelayableValidator
+      def self.included(model)
+        model.class_eval do
+          rule :parent_guid, :guid
```
Comment (SuperTux88) [inline lib/diaspora_federation/validators/relayable_validator.rb]: what about the `parent_author_signature` and `author_signature`?

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: The only applied rule was "not_empty", which actually isn't true. There are [no restriction](https://github.com/diaspora/diaspora/blob/develop/db/schema.rb#L170-L171) in DB schema and ```parent_author_signature``` is actually nil in [some](https://github.com/diaspora/diaspora/blob/develop/lib/diaspora/relayable.rb#L106-L109) circumstances.

File: spec/lib/diaspora_federation/validators/comment_validator_spec.rb (hunk 1 above)
Comment (cmrd-senya) [inline spec/lib/diaspora_federation/validators/comment_validator_spec.rb]: Done

Comment (cmrd-senya) [inline spec/lib/diaspora_federation/validators/comment_validator_spec.rb]: But for ```parent_guid``` only, again.

File: lib/diaspora_federation/validators/relayable_validator.rb (hunk 2 above)
Comment (SuperTux88) [inline lib/diaspora_federation/validators/relayable_validator.rb]: But I think at least `author_signature` should be there or the signature-verification will fail?

I don't know where the `parent_author_signature` is used and what happens if it is empty.

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: The code in Diaspora is not very clear, I think I should test this part more close to be sure.

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: Actually my tests show an opposite thing: parent_author_signature is mandatory and author_signature is optional. Here is logs:
[Both parameters are empty](https://gist.github.com/cmrd-senya/874abcaf75254e39d7cc#file-both-empty-log-L14).
[Both parameters are filled](https://gist.github.com/cmrd-senya/3212e954eec9801424da#file-both-filled-log-L10).
[Only author_signature is empty](https://gist.github.com/cmrd-senya/cd5669b86b598cd9554b#file-author_signature-empty-log-L10).
[Only parent_author_signature is empty](https://gist.github.com/cmrd-senya/d9b7bcad9c583bed1888#file-parent_author_signature-empty-log-L10).

Maybe that means a bug in Diaspora?

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: It looks like ```author_signature``` never gets checked really. The only thing that works for the relayable ancestors is ```parent_author_signature```.

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: @jhass, could you please check this outcomes? If it is a bug, we need to find out, how it is supposed to work.

Comment (jhass) [inline lib/diaspora_federation/validators/relayable_validator.rb]: I think you're probably deeper into the related code atm that I am. Did you compare behaviour between

* parent author is local
* parent author is remote
* author is local
* author is remote (received object as pod hosting the parent author)

Comment (cmrd-senya) [inline lib/diaspora_federation/validators/relayable_validator.rb]: I only ran test for a case when author and parent author is remote. But I don't see any place in the source code where ```author_signature``` check affects execution flow at all.

Actually, I can write some integration tests for this, if I know what is the idea behind it.

**Comment by SuperTux88** [comment]
I merged this now, but we need to have a look at the signature-stuff again. But I think you can do this in #13.

**Pull request merged**